src/main/java/de/tum/in/www1/artemis/domain/BaseExercise.java

```diff
@@ -177,14 +177,7 @@ public void setExampleSolutionPublicationDate(@Nullable ZonedDateTime exampleSol
      *
      * @return true if example solution publication date is in the past, false otherwise (including null case).
      */
-    public boolean isExampleSolutionPublished() {
-        if (this.isExamExercise()) {
-            // This feature is currently not available for exam exercises, this should return false
-            // for exam exercises until the conditions for them is fully implemented.
-            return false;
-        }
-        return this.exampleSolutionPublicationDate != null && ZonedDateTime.now().isAfter(this.exampleSolutionPublicationDate);
-    }
+    public abstract boolean isExampleSolutionPublished();
```

Comment:
Do we even need this abstract method here?

Reply:
The unchanged version had the method declared and defined in `BaseExercise.java` so I kept it that way. After your comment I checked the usages and I couldn't see anything preventing us from moving it to `Exercise.java` so I removed the abstract method declaration.